src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -481,15 +481,32 @@ class ApplicationManagerImpl
    *
    * @return true on success, false if passthru is already in process
    */
+  // DEPRECATED
   bool BeginAudioPassThrough() OVERRIDE;
 
+  /**
+   * @brief Starts AudioPassThru process by given application
+   * @param session_key Session key of connection for Mobile side
+   * @return true if AudioPassThru can be started, false otherwise
+   */
+  bool BeginAudioPassThru(int32_t session_key) OVERRIDE;
+
   /*
    * @brief Finishes already started audio passthru process
    *
    * @return true on success, false if passthru is not active
    */
+  // DEPRECATED
   bool EndAudioPassThrough() OVERRIDE;
 
+  /**
+   * @brief Finishes already started AudioPassThru process by given application
+   * @param application_key ID of the application which started the process
+   * @return true if AudioPassThru process has been started with given
+   * application and thus it can be stopped, false otherwise
+   */
+  bool EndAudioPassThru(int32_t application_key) OVERRIDE;
```

Comment:
Argument name should be unified, perhaps `session_key` in both cases.

Reply:
My intention was to sync with application_manager.h where its `StartAudioPassThruThread()` takes `session_key` and `StopAudioPassThru()` takes `application_key`. Eventually they are same thing and I'm OK to unify the names. Please refer to the next comment as well.